nucleus/errors.py

```diff
@@ -22,3 +22,12 @@ class DatasetItemRetrievalError(Exception):
     def __init__(self, message="Could not retrieve dataset items"):
         self.message = message
         super().__init__(self.message)
+
+
+class NucleusAPIError(Exception):
```

Comment:
In the future, I think it would be really productive to overhaul error handling on the server side to return a sentry error_id here (similar to 500 errors in the UI) such that when a user reports a bug we can lookup the stack trace

Reply:
That would be nice!